rust/worker/src/blockstore/arrow/blockfile.rs

```diff
@@ -408,6 +564,169 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn test_prefix() {
+        let tmp_dir = tempfile::tempdir().unwrap();
+        let storage = Storage::Local(LocalStorage::new(tmp_dir.path().to_str().unwrap()));
+        let blockfile_provider = ArrowBlockfileProvider::new(storage);
+        let writer = blockfile_provider.create::<&str, u32>().unwrap();
+        let id = writer.id();
+
+        // write 2000 rows per prefix for 5 prefixes.
+        let num_keys = 2000;
+        for j in 1..=5 {
+            let prefix = format!("{}/{}", "prefix", j);
+            for i in 1..=num_keys {
+                let key = format!("{}/{}", "key", i);
+                writer
+                    .set(prefix.as_str(), key.as_str(), i as u32)
+                    .await
+                    .unwrap();
+            }
+        }
+        // commit.
+        writer.commit::<&str, u32>().unwrap();
+
+        let reader = blockfile_provider.open::<&str, u32>(&id).await.unwrap();
+        // Generate a random number between 1 and 5.
+        let mut rng = rand::thread_rng();
```

Comment:
for my understanding - will this respect proptest seed?

Reply:
No this won't. Good catch - I should also update this test to use proptest